hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -482,11 +513,29 @@ extends Importer:
                   case ((pss, ctx), ps) => 
                     val (qs, newCtx) = params(ps)(using ctx)
                     (pss :+ ParamList(ParamListFlags.empty, qs), newCtx)
+              // * Elaborate signature
+              val s = td.signature.orElse(newSignatureTrees.get(id.name)).map(term)
               val b = rhs.map(term(_)(using newCtx))
               val r = FlowSymbol(s"‹result of ${sym}›", nextUid)
-              val tdf = TermDefinition(owner, k, sym, pss,
-                td.signature.orElse(newSignatureTrees.get(id.name)).map(term), b, r)
+              val tdf = TermDefinition(owner, k, sym, pss, s, b, r, TermDefFlags(mod))
               sym.defn = S(tdf)
+
+              // Restrictions regarding functions receiving module parameters
+              
+              // not working because sometimes the signature is N 
+              // even when the function has a explicit return type
+              // val explicitReturnType = s.isDefined
+              
+              // FIXME: this is magic; we shall determine the return type based on the signature
```

Comment:
So is this FIXME meant to be addressed in this PR?

Reply:
This makes me realize we haven't discussed how to mark module returns. Following my latest proposal for module parameters, we should probably make it:

```fs

module fun example(module m: Foo, x, y): Bar = ...
```

WDYT?